packages/nextjs/src/utils/serverIntegrations.ts

```diff
@@ -0,0 +1,54 @@
+import { RewriteFrames } from '@sentry/integrations';
+import { Integration } from '@sentry/types';
+
+const SOURCEMAP_FILENAME_REGEX = /^.*\/.next\//;
+
+/** Default RewriteFrames integration to match filenames in Sentry. */
+export const defaultRewriteFrames = new RewriteFrames({
+  iteratee: frame => {
+    frame.filename = frame.filename?.replace(SOURCEMAP_FILENAME_REGEX, 'app:///_next/');
+    return frame;
+  },
+});
+
+/**
+ * Retrieves the patched integrations for the server.
+ *
+ * There must be a RewriteFrames integration.
+ * If the user has defined one, there's nothing to patch.
+ * If not, the default RewriteFrames integration is used.
+ *
+ * @param userIntegrations integrations defined by the user.
+ * @returns final integrations, patched if necessary.
+ */
+export function getFinalServerIntegrations(
+  userIntegrations: Integration[] | ((integrations: Integration[]) => Integration[]),
+): Integration[] | ((integrations: Integration[]) => Integration[]) {
+  if (Array.isArray(userIntegrations)) {
+    return getFinalIntegrationArray(userIntegrations);
+  } else {
+    return getFinalIntegrationFunction(userIntegrations);
+  }
+}
+
+/** Returns the patched integrations array. */
+function getFinalIntegrationArray(userIntegrations: Integration[]): Integration[] {
+  const rewriteFramesIntegration = defaultRewriteFrames;
+  if (userIntegrations.map(integration => integration.name).includes(rewriteFramesIntegration.name)) {
+    return userIntegrations;
+  } else {
+    userIntegrations.push(rewriteFramesIntegration);
+    return userIntegrations;
+  }
+}
```

Comment:
I think this can be simplified a bit, which would make it easier to read.

```suggestion
function getFinalIntegrationArray(userIntegrations: Integration[]): Integration[] {
  if (userIntegrations.map(integration => integration.name).includes('RewriteFrames')) {
    return userIntegrations;
  } 
    
  userIntegrations.push(defaultRewriteFrames);
  return userIntegrations;
}
```

(If you want to get super fancy with it 😛 , you could also do

```
function getFinalIntegrationArray(userIntegrations: Integration[]): Integration[] {
  return (
    userIntegrations.map(integration => integration.name).includes('RewriteFrames') 
    ? userIntegrations 
    : userIntegrations.concat([defaultRewriteFrames])
  );
}
```

but I think doing it with the `if` is totally fine, too.)

In any case, there are a number of other places where you have unnecessary `else`s, and IMHO as a rule 

```
if(x) {
  return someValue;
}
                                    <--- notice the blank line to visually separate the cases
return someOtherValue;
```

is always going to be more digestible than 

```
if(x) {
  return someValue;
} else {
  return someOtherValue;
}
```

especially if there's a bunch of code in either block.

Reply:
Removing the unnecessary `else`s and tmp variable assignment; but not hardcoding `"RewriteFrames"` in the `if`, it's less solid IMO. Also, not using `concat` to avoid creating a new array.

Reply:
> not hardcoding "RewriteFrames" in the if, it's less solid IMO

Up to you. 🙂  And as an absolute principle, you are right. However, the chances of us ever changing the name are slim to none, and if we ever did, we'd comb through our code and change it everywhere. Worth the increase in readability, IMHO, but your choice.